projects/sensory_interface/public/app.js

```diff
@@ -182,4 +184,28 @@ function findMinAndMaxValues() {
     }
     document.getElementById("maxValue").value = maxValue;
     document.getElementById("minValue").value = minValue;
+}
+
+function updateURL() {
+    let input = document.getElementById("dataInput").value;
+    let URLDataPart = "data=" + input;
+    URLDataPart = encodeURIComponent(URLDataPart);
+    window.history.pushState({
+        state: "examining data"
+    },
+        "",
+        "/index.html?" + URLDataPart);
+}
+
+function getDataFromURL() {
+    let URL = window.location.href;
```

Comment:
Try using `searchParams`: https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams

Reply:
Thanks so much for this Ofer!! This made the code much prittier.

Reply:
Sure thing :))